src/fileformat/file_format/elf/elf_format.cpp

```diff
@@ -1744,10 +1745,18 @@ void ElfFormat::loadSymbols(const ELFIO::elfio *file, const ELFIO::symbol_sectio
 		telfhashSymbols = {};
 	}
 
+	// Cache keeping track of created imports (i.e. (name, address) pairs)
+	// to prevent repeated creation of the same imports.
+	// To save space, this uses string references and relies on imports not
+	// being moved -- they should not be, as ImportTable stores vector of unique
+	// pointers, but if that ever changes, this will end badly.
+	std::set<std::pair<const std::string&, unsigned long long>> createdImports;
```

Comment:
Honestly if I were to cache addresses I'd choose pointers to denote that. I understand it works with references but disadvantage here is simply the fact that references are automatically dereferenced in C++. It would be fairly easy to make a mistake if this code is going to be modified in the future, as you might easily unknowingly dereference something that you didn't mean to. Pointer explicitly forces you to take the address of the value you want to store, so you always know what you are working with and it's harder to make a mistake.

Reply:
Ok, I had to write custom comparator, but you are right that it is a bit more safe.